stack-snapshot.yaml

```diff
@@ -5,6 +5,7 @@ resolver: nightly-2022-01-26
 packages:
   - aeson-2.0.3.0
   - bytestring-nums-0.3.6
+  - entropy-0.4.1.10
```

Comment:
when I first repinned the windows stackage snapshot (after bumping `optparse-applicative`), I noticed that we ended up with a different revision of `entropy`

> previously it was `entropy-0.4.1.7@rev0`
> (`sha256:56ebebf726696e204e5126d748defdfc8453ff9d0a817b21a704d92e7bad5f18`) 
> after repinning it was `entropy-0.4.1.7@rev2`
> (`sha256:6c09f55d7fe74002f5e90cb433650c442476b066b89cb01aceb1d2dd61901cc0`) 
> \- see https://hackage.haskell.org/package/entropy-0.4.1.7/revisions/

`rev2` is effectively useless on Windows since it sets the `Win32` dependency bounds to `<0` (IIUC this was done to avoid build errors on GHC >=9.6.1 - https://github.com/haskell/entropy/issues/80, so it shouldn't affect us), so I tried to pin it to `rev0` on this file and on `bazel-haskell-deps.bzl` in different ways but I couldn't get it to pick that one. (Later I'll try to make a minimal example to raise as a rules-haskell issue).

Since I wasn't able to do it, and given that the only package we use that depends directly on `entropy` (`uuid`) only sets an upper bound of `<0.5` on `entropy`, I thought it would be fine to bump it to the latest version `entropy-0.4.1.10`, which does compile on Windows.